testing/internal/e2e/tests/base/role_multi_scope_grants_test.go

```diff
@@ -0,0 +1,165 @@
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+	newProjectId := boundary.CreateNewProjectCli(t, ctx, newOrgId)
+
+	// Create Account
+	acctName := "e2e-account"
+	newAccountId, acctPassword := boundary.CreateNewAccountCli(t, ctx, bc.AuthMethodId, acctName)
+	t.Cleanup(func() {
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary",
+			e2e.WithArgs("accounts", "delete", "-id", newAccountId),
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+
+	// Create User and set Account to it
+	newUserId := boundary.CreateNewUserCli(t, ctx, "global")
+	t.Cleanup(func() {
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary",
+			e2e.WithArgs("users", "delete", "-id", newUserId),
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+	boundary.SetAccountToUserCli(t, ctx, newUserId, newAccountId)
+
+	// Create Role and add admin grants to it
+	newRoleId, err := boundary.CreateRoleCli(t, ctx, "global")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary",
+			e2e.WithArgs("roles", "delete", "-id", newRoleId),
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+	boundary.AddGrantToRoleCli(t, ctx, newRoleId, "ids=*;type=*;actions=*")
+
+	// Authenticate test user and try to:
+	// - create role in global scope: expect error
+	// - create role in org scope: expect error
+	// - create role in proj scope: expect error
+	boundary.AuthenticateCli(t, ctx, bc.AuthMethodId, acctName, acctPassword)
+
+	_, err = boundary.CreateRoleCli(t, ctx, "global")
+	require.Error(t, err)
+
+	_, err = boundary.CreateRoleCli(t, ctx, newOrgId)
+	require.Error(t, err)
+
+	_, err = boundary.CreateRoleCli(t, ctx, newProjectId)
+	require.Error(t, err)
+
+	// Add User as a principle to the admin Role
+	boundary.AuthenticateAdminCli(t, ctx)
+	boundary.AddPrincipalToRoleCli(t, ctx, newRoleId, newUserId)
+
+	// Authenticate User and try to:
+	// - create Role in global scope: expect success
+	// - create Role in org scope: expect error
+	// - create Role in proj scope: expect error
+	boundary.AuthenticateCli(t, ctx, bc.AuthMethodId, acctName, acctPassword)
+	newGlobalRoleId2, err := boundary.CreateRoleCli(t, ctx, "global")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary",
+			e2e.WithArgs("roles", "delete", "-id", newGlobalRoleId2),
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+
+	_, err = boundary.CreateRoleCli(t, ctx, newOrgId)
+	require.Error(t, err)
+
+	_, err = boundary.CreateRoleCli(t, ctx, newProjectId)
+	require.Error(t, err)
+
+	// Set Grant Scopes to Role: this, children
+	boundary.AuthenticateAdminCli(t, ctx)
+	err = boundary.SetGrantScopesToRoleCli(t, ctx, newRoleId,
+		boundary.WithGrantScopeId("this"),
```

Comment:
I didn't know `this` was a valid `grant-scope-id`... what if you didn't have `this` and just `children`?

Reply:
Then the grants will apply only in children scopes and not in the role's scope